bigquery/v2/query/reader.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package query
+
+import (
+	"context"
+
+	"cloud.google.com/go/bigquery/storage/apiv1/storagepb"
+	"cloud.google.com/go/bigquery/v2/apiv2/bigquerypb"
+)
+
+// Reader is used to read the results of a query.
```

Comment:
we have a Query, a Reader, and a RowIterator?  This feels like an overlap at a high level, but we can dig in when we talk next.

Reply:
just Query and RowIterators are going to be public, pushed some changes to outline that. `Reader` is an interface for reading data using `jobs.query` or using Storage Read API.